app/Services/Image/ImageUploadService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Services\Image;
+
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
+use Symfony\Component\HttpFoundation\File\UploadedFile as SymfonyUploadedFile;
+
+/**
+ * Service class for handling image uploads.
+ */
+class ImageUploadService
+{
+    /**
+     * Upload a file to the specified destination path with an optional prefix.
+     *
+     * @param \Illuminate\Http\UploadedFile $file
+     * @param string $destinationPath
+     * @param string $prefix
+     * @return string
+     * @throws \App\Exceptions\ImageUploadException
+     */
+    public function uploadImage($file, $destinationPath, string $prefix = '')
+    {   
+        if (!$file->isValid()) {
+            $notification = [
+                'message' => 'Error Uploading Image File',
+                'alert-type' => 'error',
+            ];
+            return redirect()->back()->with($notification);
+        }
```

Comment:
The validation (or at least the redirection part) should not be a responsibility of the service.